### PR TITLE
#20 Enemy Ranged Attack

### PR DESCRIPTION
**Pull request by joshjwilli**
- Enemy prefab has a ranged attack now, works as described in the issue
- Created Enemy Projectile Prefab, works as described in the issue
- Created general Projectile Script which should work for player or enemy projectiles
- Projectile Stats are handled through the projectile script (meaning per enemy/player stats control is not possible, a new prefab/variation of a prefab would need to be made for that kind of thing)

### CODE REVIEW

File: Assets/Scripts/EnemyController.cs
Comment (adenprince) [inline Assets/Scripts/EnemyController.cs]: Suggested naming changes:
- `proj` -> `projectile`
  - It is not clear what `proj` means just by looking at the instance variables in this script. I'm not sure if the `ProjController` class should be renamed to `ProjectileController`, but I think `ProjectileController` would be better.
- `playerPos` -> `playerTransform`
  - Position is just a property of the transform, so `playerTransform` is more accurate. (I think `player` would also be fine.)

File: Assets/Scripts/EnemyController.cs
Comment (adenprince) [inline Assets/Scripts/EnemyController.cs]: This `Debug.Log` call can be removed.

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: I don't think there's much benefit to giving these instance variables default values because they will be set in the Inspector.

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: An empty string should be used instead of the string `"null"` because it is confusing to use that when a string can have a value of `null`. (I don't think a string can be set to `null` in the Inspector.)

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: Rename `targetPos` to `targetTransform` (or `target`)

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: `rb` and `moveDirect` can be local variables in the `Start` method. Also, I think `moveDirect` should be renamed to `velocity`.

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: I would prefer calculating a deviation vector and then adding that to `moveDirect`, but this way is also fine with me.

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: `Debug.LogError` can be used here.

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: I think an error should be logged when `player` (or `enemy` below) is null, but it doesn't matter that much.

**Review by adenprince — COMMENTED** [review]
Looks good. All of my comments are for small changes that I would like to be considered before this branch is merged.

File: Assets/Scripts/ProjController.cs
Comment (joshjwilli) [reply Assets/Scripts/ProjController.cs]: Is there a reason to do that, beyond personal preference?

File: Assets/Scripts/ProjController.cs
Comment (joshjwilli) [reply Assets/Scripts/ProjController.cs]: Fair enough, but also what are the drawbacks of having default values?

Comment (adenprince) [reply Assets/Scripts/ProjController.cs]: There's not really a drawback in this case. I usually avoid putting in default values due to the behavior described in [this post](https://stackoverflow.com/a/51558490), but the functionality described there doesn't really apply here. I feel like the default values are kind of unnecessary, but it might be good to have baseline projectile values as defaults. There's no real need to remove the default values.

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [reply Assets/Scripts/ProjController.cs]: I think it simplifies the code a bit. It also makes it easier to extract deviation calculation into a method, but that probably isn't something we'll need to do. You can leave it as is.

File: Assets/Scripts/ProjController.cs
Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: To clarify why I requested `proj` be changed to `projectile`: I think it's clear in this class what `proj` is short for (because the class is called `ProjectileController`), but it might be unclear what `proj` means when it's not close to the word projectile.

Comment (adenprince) [inline Assets/Scripts/ProjController.cs]: - [x] Remove `ProjController.cs`

File: Assets/Scripts/ProjectileController.cs
Comment (adenprince) [inline Assets/Scripts/ProjectileController.cs]: Variables is misspelled here.

**Review by adenprince — CHANGES_REQUESTED** [review]
`ProjController.cs` needs to be removed. I will approve once that file is removed.

File: Assets/Scripts/ProjController.cs
Comment (joshjwilli) [reply Assets/Scripts/ProjController.cs]: Done!

File: Assets/Scripts/ProjectileController.cs
Comment (adenprince) [inline Assets/Scripts/ProjectileController.cs]: I think this method should be structured like this (comments are just placeholders):
```
if (targetName == "Player")
{
    // ...
}
else if (targetName == "Enemy")
{
    // ...
}
else
{
    // Error
}
```
Then, there's an error when `targetName` is not `null`, `"Player"`, or `"Enemy"`. Currently, this method does nothing in that case. Erroring in that case makes it clearer that `targetName` has an invalid value.

**Review by adenprince — APPROVED** [review]

**Comment by adenprince** [comment]
Looks good. Thank you for quickly making changes.

**Pull request merged**
